File: 179.clj
(defn make-divs-seq [limit]
  (loop [i (int 0), d (int 2), divs (int-array limit 1)]
    (cond
      (== d limit) divs
      (>= i limit) (recur (int 0) (unchecked-inc d) divs)
      :default (recur (unchecked-add i d)
                      d
                      (do
                        (aset divs i (unchecked-inc (aget divs i)))
                        divs)))))

(loop [divs-seq (make-divs-seq (int 1e7)), same-cnt 0]
  (if (== 1 (count divs-seq))
    (println same-cnt)
    (recur
      (rest divs-seq)
      (if (= (first divs-seq) (second divs-seq)) (inc same-cnt) same-cnt))))

File: 179.rb
#!/usr/bin/ruby
# vi:ts=2:sw=2

LIMIT = 10**7

divs = Array.new(LIMIT, 1)

2.upto(LIMIT) do |d|
	i = 0
	while i < LIMIT
		divs[i] += 1
		i += d
	end
end

i = count = 0
while (i += 1) < LIMIT - 2
	if divs[i] == divs[i + 1]
		count += 1
	end
end

puts count
